test/e2e/kubevirt.go

```diff
@@ -1752,23 +1752,22 @@ write_files:
 			iperfServerTestPods, err = createIperfServerPods(selectedNodes, cudn.Name, td.role, []string{})
 			Expect(err).NotTo(HaveOccurred())
 
-			network, err := infraprovider.Get().PrimaryNetwork()
-			Expect(err).ShouldNot(HaveOccurred(), "primary network must be available to attach containers")
-			if containerNetwork := containerNetwork(td); containerNetwork != network.Name() {
-				network, err = infraprovider.Get().GetNetwork(containerNetwork)
-				Expect(err).ShouldNot(HaveOccurred(), "must to get alternative network")
-			}
-			externalContainerPort := infraprovider.Get().GetExternalContainerPort()
-			externalContainerName := namespace + "-iperf"
-			externalContainerSpec := infraapi.ExternalContainer{
-				Name:    externalContainerName,
-				Image:   images.IPerf3(),
-				Network: network,
-				Args:    []string{"sleep infinity"},
-				ExtPort: externalContainerPort,
+			var externalContainer infraapi.ExternalContainer
+			if td.role == udnv1.NetworkRolePrimary {
+				primaryProviderNetwork, err := infraprovider.Get().PrimaryNetwork()
+				Expect(err).ShouldNot(HaveOccurred(), "primary network must be available to attach containers")
+				externalContainerPort := infraprovider.Get().GetExternalContainerPort()
+				externalContainerName := namespace + "-iperf"
+				externalContainerSpec := infraapi.ExternalContainer{
+					Name:    externalContainerName,
+					Image:   images.IPerf3(),
+					Network: primaryProviderNetwork,
+					Args:    []string{"sleep infinity"},
+					ExtPort: externalContainerPort,
+				}
+				externalContainer, err = providerCtx.CreateExternalContainer(externalContainerSpec)
```

Comment:
commit title seems a bit off - youve adapted the tests to create an external container here.